### PR TITLE
bump kube-rbac-proxy

### PR DESCRIPTION
**Pull request by joshmeranda**
#### Pull Request Checklist ####

- [x] Change does not remove any existing Images or Tags in the images-list file
- [x] Change does not remove / overwrite exiting Images or Tags in Rancher DockerHub
- [x] If updating an existing entry, verify the `SOURCE` is still accurate and upstream hasn't been migrated to a new regitry or repo (if they've migrated, a new repo request to EIO is needed to comply with the `SOURCE DESTINATION TAG` pattern)
- [x] New entries are in format `SOURCE DESTINATION TAG`
- [x] New entries are added to the correct section of the list (sorted lexicographically)
- [x] New entries have a repo created in Rancher Dockerhub (where the image will be mirrored to)
- [x] New entries are licensed with Rancher favored licenses - Apache 2 and MIT - or approved licenses - as according to [CNCF approved licenses](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md).
- [x] New entries, when used in Rancher or Rancher's provided charts, have their corresponding origin added in Rancher's images [origins](https://github.com/rancher/rancher/blob/release/v2.7/pkg/image/origins.go) file (must be added for all Rancher versions `>= v2.7`).
- [x] Changes to scripting or CI config have been tested to the best of your ability

#### Types of Change ####

Version bump for kube-rbac-proxy `0.5.0` -> `0.14.0`

#### Linked Issues ####

https://github.com/rancher/rancher/issues/42019
https://github.com/rancher/charts/pull/2785

#### Final Checks after the PR is merged ####
- [x] Confirm that you can pull the new images and tags from DockerHub

### CODE REVIEW

File: images-list [hunk 1]
```diff
@@ -206,7 +206,9 @@ gcr.io/google_containers/k8s-dns-sidecar rancher/mirrored-k8s-dns-sidecar 1.15.1
 gcr.io/google_containers/pause rancher/mirrored-pause 3.1
 gcr.io/google_containers/pause rancher/mirrored-pause 3.2
 gcr.io/kubebuilder/kube-rbac-proxy rancher/kube-rbac-proxy v0.5.0
+gcr.io/kubebuilder/kube-rbac-proxy rancher/kube-rbac-proxy v0.14.0
```
Comment (brandond) [inline images-list:209]: do we need to keep mirroring new tags to the old un-mirrored repo? Can we update to the mirrored repo when switching to the new tag?

Comment (joshmeranda) [reply images-list:209]: Probably not, but I just assumed if both were included they were both in use and would need the update. Unless we have some charts referring to `rancher/kube-rbac-proxy:latest` that would need to have the newest versions, we can stop.

From the blame @rawmind0 added the original unmirrored image and @rmweir added the mirrored image. Maybe they have more context for why we have both in the image list / if we need to keep them updated.

Comment (joshmeranda) [reply images-list:209]: @brandond are you ok with only bumping the mirrored images moving forward? looks like thats what we do for the `rancher/bats-bats` and `rancher/mirrored-bats-bats`

Comment (brandond) [reply images-list:209]: yeah I think we should stop bumping tags for images without the mirrored- prefix.

Comment (joshmeranda) [reply images-list]: sick, fixed

**Review by brandond — APPROVED** [review]

**Pull request merged**